Purchases/Public/RCPurchases.m

```diff
@@ -477,13 +494,21 @@ - (void)purchaseProduct:(SKProduct *)product
     // This is to prevent the UIApplicationDidBecomeActive call from the purchase popup
     // from triggering a refresh.
     self.cachesLastUpdated = [NSDate date];
-    
-    RCDebugLog(@"makePurchase - %@", payment.productIdentifier);
-    
+
+    if (presentedOfferingIdentifier) {
+        RCDebugLog(@"makePurchase - %@ - Offering: %@", payment.productIdentifier, presentedOfferingIdentifier);
+    } else {
+        RCDebugLog(@"makePurchase - %@", payment.productIdentifier);
+    }
+
     @synchronized (self) {
         self.productsByIdentifier[payment.productIdentifier] = product;
     }
-    
+
+    @synchronized (self) {
+        self.offeringsByProductIdentifier[payment.productIdentifier] = presentedOfferingIdentifier;
```

Comment:
Slight improvement to readability would be to name this:
`presentedOfferingsByProductIdentifier`